Files/UserControls/ModernNavigationToolbar.xaml.cs

```diff
@@ -234,30 +236,29 @@ public async void CheckPathInput(ItemViewModel instance, string CurrentInput)
                         : App.CurrentInstance.FilesystemViewModel.WorkingDirectory;
                     var parentItem = await StorageFileExtensions.GetFolderWithPathFromPathAsync(workingDir);
 
-                    if (CurrentInput.Contains("%temp%")) CurrentInput = CurrentInput.Replace("%temp%", AppSettings.TempPath);
-                    if (CurrentInput.Contains("%homepath%")) CurrentInput = CurrentInput.Replace("%homepath%", AppSettings.HomePath);
-                    CurrentInput = Environment.ExpandEnvironmentVariables(CurrentInput);
+                    currentInput = StorageFileExtensions.GetPathWithoutEnvironmentVariable(currentInput);
+                    if (currentSelectedPath == currentInput) return;
 
-                    var item = await DrivesManager.GetRootFromPath(CurrentInput);
+                    var item = await DrivesManager.GetRootFromPath(currentInput);
                     try
                     {
-                        var pathToNavigate = (await StorageFileExtensions.GetFolderFromPathAsync(CurrentInput, item, parentItem)).Path;
+                        var pathToNavigate = (await StorageFileExtensions.GetFolderFromPathAsync(currentInput, item, parentItem)).Path;
```

Comment:
@soumyamahunt This should be
```cs
var pathToNavigate = (await StorageFileExtensions.GetFolderWithPathFromPathAsync(currentInput, item, parentItem)).Path;
```
`GetFolderFromPathAsync(...).Path` will be empty when browsing android/phones/mtp devices
(I know this is not obvious and should be written somewhere)